internal/services/managedapplications/managed_application_definition_resource.go

```diff
@@ -193,105 +190,99 @@ func resourceManagedApplicationDefinitionRead(d *pluginsdk.ResourceData, meta in
 	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
 	defer cancel()
 
-	id, err := parse.ApplicationDefinitionID(d.Id())
+	id, err := applicationdefinitions.ParseApplicationDefinitionID(d.Id())
 	if err != nil {
 		return err
 	}
 
-	resp, err := client.Get(ctx, id.ResourceGroup, id.Name)
+	resp, err := client.Get(ctx, *id)
 	if err != nil {
-		if utils.ResponseWasNotFound(resp.Response) {
+		if response.WasNotFound(resp.HttpResponse) {
 			log.Printf("[INFO] Managed Application Definition %q does not exist - removing from state", d.Id())
 			d.SetId("")
 			return nil
 		}
-		return fmt.Errorf("failed to read Managed Application Definition %q (Resource Group %q): %+v", id.Name, id.ResourceGroup, err)
+		return fmt.Errorf("failed to read Managed Application Definition %s: %+v", id, err)
 	}
 
-	d.Set("name", id.Name)
-	d.Set("resource_group_name", id.ResourceGroup)
-	d.Set("location", location.NormalizeNilable(resp.Location))
+	if m := resp.Model; m != nil {
+		p := m.Properties
+
+		d.Set("name", m.Name)
+		d.Set("resource_group_name", id.ResourceGroupName) // missing from response?
+		d.Set("location", location.NormalizeNilable(m.Location))
 
-	if props := resp.ApplicationDefinitionProperties; props != nil {
-		if err := d.Set("authorization", flattenManagedApplicationDefinitionAuthorization(props.Authorizations)); err != nil {
+		if err := d.Set("authorization", flattenManagedApplicationDefinitionAuthorization(p.Authorizations)); err != nil {
 			return fmt.Errorf("setting `authorization`: %+v", err)
 		}
-		d.Set("description", props.Description)
-		d.Set("display_name", props.DisplayName)
-		d.Set("package_enabled", props.IsEnabled)
-		d.Set("lock_level", string(props.LockLevel))
-	}
+		d.Set("description", p.Description)
+		d.Set("display_name", p.DisplayName)
+		d.Set("package_enabled", p.IsEnabled)
+		d.Set("lock_level", string(p.LockLevel))
+
+		// the following are not returned from the API so lets pull it from state
+		if v, ok := d.GetOk("create_ui_definition"); ok {
+			d.Set("create_ui_definition", v.(string))
+		}
+		if v, ok := d.GetOk("main_template"); ok {
+			d.Set("main_template", v.(string))
+		}
+		if v, ok := d.GetOk("package_file_uri"); ok {
+			d.Set("package_file_uri", v.(string))
+		}
 
-	// the following are not returned from the API so lets pull it from state
-	if v, ok := d.GetOk("create_ui_definition"); ok {
-		d.Set("create_ui_definition", v.(string))
-	}
-	if v, ok := d.GetOk("main_template"); ok {
-		d.Set("main_template", v.(string))
-	}
-	if v, ok := d.GetOk("package_file_uri"); ok {
-		d.Set("package_file_uri", v.(string))
+		return tags.FlattenAndSet(d, m.Tags)
```

Comment:
pedantic, but can we make this:

```suggestion
		if err := tags.FlattenAndSet(d, model.Tags); err != nil {
			return fmt.Errorf("setting `tags`: %+v", err)
		}
```

so that we have a single exit out of this function? Again it's minor, but doing so means we don't unintentionally miss a code path if something gets added below this in a subsequent PR review